overlord/snapstate/handlers.go

```diff
@@ -1906,7 +1905,9 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) (err error) {
 		}
 	}
 	if !rebootInfo.RebootRequired || canReboot {
-		m.maybeRestart(t, newInfo, rebootInfo)
+		if err := m.finishTaskWithMaybeRestart(t, state.DoneStatus, newInfo, rebootInfo); err != nil {
```

Comment:
I'm confused by these calls (probably because I don't know the background and what the `status` parameter is really going to be used for), but when I read this line it makes me think that the `finishTaskWithMaybeRestart` might trigger a restart and "finish" the task by setting it to `state.DoneStatus`. But then I noticed that we already have a line

    t.SetStatus(state.DoneStatus)

above, so I guess that my initial understanding is not right...

Reply:
that was my original proposal to make the function deal with that setting as well if called, Alfonso was worried that we would have two styles: tasks that call this and task that don't and use SetStatus directly. Your wondering make me think that my initial thinking wasn't too off and the repetition about the status is confusing in itself and not great, as the two values could get out of sync as well.

Reply:
Changed finally so the status is set by FinishTaskWithRestart